backend/elasticsearch/pom.xml

```diff
@@ -56,4 +56,30 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.moditect</groupId>
+                <artifactId>moditect-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>add-module-infos</id>
+                        <configuration>
+                            <module>
+                                <moduleInfo>
+                                    <exports combine.self="override">
+                                        org.hibernate.search.*.logging.impl to org.jboss.logging;
+                                        org.hibernate.search.*.reporting.impl to org.jboss.logging;
+                                        org.hibernate.search.*.impl to org.hibernate.search.engine, gson;
```

Comment:
I suspect that what we need for `gson` is probably an `opens` statements instead of an `exports` statements. Because `gson` may need reflection and access to private fields, which `exports` doesn't provide.

I suspect the only reason it currently works with just an `exports` is that we use our own type adapters (e.g. `PropertyMappingJsonAdapterFactory`) on all the serialized types. 

Would you mind replacing the export to `gson` with something like this? It will probably be more future-proof.

```
<opens>
org.hibernate.search.backend.elasticsearch.lowlevel.index.* to gson
</opens>
```

Reply:
Good idea!